crates/payload/basic/src/lib.rs

```diff
@@ -715,7 +834,10 @@ fn build_payload<Pool, Client>(
             gas_limit: block_gas_limit,
             difficulty: U256::ZERO,
             gas_used: cumulative_gas_used,
+            #[cfg(not(feature = "optimism"))]
             extra_data: extra_data.into(),
+            #[cfg(feature = "optimism")]
+            extra_data: if chain_spec.optimism { Default::default() } else { extra_data.into() },
```

Comment:
It would be cool if we could push this branching into the `PayloadConfig`.

ie, `PayloadConfig` has a `pub fn extra_data(&self) -> Bytes` that would look something like:
```
impl PayloadConfig {
    pub fn extra_data(&self) -> &Bytes {
        #[cfg(feature = "optimism")]
        if chain_spec.optimism {
            Default::default()
        }
        self.extra_data
    }
}
```

Then, here we could just do:
```suggestion
            extra_data: config.extra_data().into(),
```